Objects/listobject.c

```diff
@@ -2886,6 +2898,33 @@ listiter_setstate(listiterobject *it, PyObject *state)
     Py_RETURN_NONE;
 }
 
+static int
+listiter_contains(listiterobject *it, PyObject *el)
+{
+    PyListObject *seq;
+    Py_ssize_t i;
+    int cmp;
+
+    assert(it != NULL);
+    seq = it->it_seq;
+    if (seq == NULL)
+        return 0;
+    assert(PyList_Check(seq));
+
+    for (i = it->it_index, cmp = 0 ; cmp == 0 && i < Py_SIZE(seq); ++i)
+        cmp = PyObject_RichCompareBool(el, PyList_GET_ITEM(seq, i),
```

Comment:
`seq` is a borrowed reference. `PyObject_RichCompareBool()` can call a user code which can exhaust the list iterator. This can make `seq` a hanging reference.